mrp_lot_cost/models/mrp.py

```diff
@@ -0,0 +1,75 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    @authors: Alexander Ezquevo <alexander@acysos.com>
+#    Copyright (C) 2015  Acysos S.L.
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api
+
+
+class Lot(models.Model):
+    _inherit = 'stock.production.lot'
+
+    unit_cost = fields.Float(string='Lot unit cost')
+
+
+class MRP_production(models.Model):
+    _inherit = 'mrp.production'
+
+    @api.multi
+    def action_produce(
+            self, production_qty, production_mode,
+            wiz=False):
+        if wiz and wiz.lot_id and len(wiz.consume_lines) > 0:
+            control = True
+            for line in wiz.consume_lines:
+                if not line.lot_id:
+                    control = False
+            if(control):
+                self.calculateCost(wiz)
+        super(MRP_production, self).action_produce(
+            self.id, production_qty, production_mode, wiz)
+
+    @api.one
```

Comment:
api.one is deprecated in 9. To help forward compatibility please use api.multi + ensure_one
see https://www.odoo.com/documentation/8.0/reference/orm.html#openerp.api.one